backend/src/utils/groupsUtils.ts

```diff
@@ -96,8 +96,8 @@ export const getAllGroupsByUser = async (
       'groups',
     );
     const groups = adminGroupResponse.body as GroupCustomObject;
-    return groups.items.filter((x) => x.users.includes(username)).map((x) => x.metadata.name);
+    return groups.items.filter((x) => x.users?.includes(username)).map((x) => x.metadata?.name);
```

Comment:
If the `users` are not always an array and could be `undefined`, could you also add this to the `GroupCustomObjectItem` type `users?: string[];`. And the same for `metadata`, if it's always defined, you may not need the `?` but if it could be `undefined` you may also want to change its type definition.

Reply:
Yes, i thought about it tx! I'll add it